src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -119,52 +126,33 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
     var_types      targetType  = node->TypeGet();
     var_types      baseType    = node->gtSIMDBaseType;
 
-    regNumber op1Reg = op1->gtRegNum;
-    regNumber op2Reg = REG_NA;
-    emitter*  emit   = getEmitter();
+    instruction ins    = INS_invalid;
+    regNumber   op1Reg = op1->gtRegNum;
+    regNumber   op2Reg = REG_NA;
+    emitter*    emit   = getEmitter();
 
     genConsumeOperands(node);
 
     switch (intrinsicID)
     {
         case NI_SSE2_Add:
-        {
-            op2Reg = op2->gtRegNum;
+            assert((baseType == TYP_BYTE) || (baseType == TYP_UBYTE) ||
+                   (baseType == TYP_SHORT) || (baseType == TYP_USHORT) ||
+                   (baseType == TYP_INT) || (baseType == TYP_UINT) ||
+                   (baseType == TYP_LONG) || (baseType == TYP_ULONG) ||
+                   (baseType == TYP_DOUBLE));
 
-            instruction ins;
-            switch (baseType)
-            {
-                case TYP_DOUBLE:
-                    ins = INS_addpd;
-                    break;
-                case TYP_INT:
-                case TYP_UINT:
-                    ins = INS_paddd;
-                    break;
-                case TYP_LONG:
-                case TYP_ULONG:
-                    ins = INS_paddq;
-                    break;
-                case TYP_BYTE:
-                case TYP_UBYTE:
-                    ins = INS_paddb;
-                    break;
-                case TYP_SHORT:
-                case TYP_USHORT:
-                    ins = INS_paddw;
-                    break;
-                default:
-                    unreached();
-                    break;
-            }
+            ins    = compiler->insOfHWIntrinsic(intrinsicID, baseType);
+            op2Reg = op2->gtRegNum;
```

Comment:
Seems like this `op2Reg` variable and the corresponding assignment is pointless.

Reply:
Not always - for Sse2.Insert there are 3 arguments Vector128, short/ushort, immediate.
Avx and above use almost always 3 args.

Reply:
I don't understand what that has to do with my comment.

Reply:
I can change it to pass `op2Reg` directly for the binary instructions.

Reply:
Fixed.